src/datatrove/pipeline/filters/url_filter.py

```diff
@@ -53,22 +53,32 @@ def __init__(
         banned_words: Iterable = None,
         banned_subwords: Iterable = None,
         soft_banned_words: Iterable = None,
+        use_integrated_lists: bool = True,
         exclusion_writer: DiskWriter = None,
     ):
+        import ahocorasick
         from tldextract import TLDExtract
 
         super().__init__(exclusion_writer)
         self.soft_word_threshold = soft_word_threshold
-        self.block_listed_domains = extra_domains
-        self.block_listed_url = extra_urls
-        self.banned_words = banned_words
-        self.banned_subwords = banned_subwords
-        self.soft_banned_words = soft_banned_words
+        self.block_listed_domains = parse_list(extra_domains, do_normalize=False) if extra_domains else set()
+        self.block_listed_url = parse_list(extra_urls, do_normalize=False) if extra_urls else set()
+        self.banned_words = parse_list(banned_words) if banned_words else set()
+        self.banned_subwords = parse_list(banned_subwords) if banned_subwords else set()
+        self.soft_banned_words = parse_list(soft_banned_words) if soft_banned_words else set()
+        self.use_integrated_lists = use_integrated_lists
         self._downloaded = False
         self.tldextractor = TLDExtract()
 
+        self.banned_subwords_automaton = ahocorasick.Automaton(ahocorasick.STORE_INTS)
+        for word in self.banned_subwords:
```

Comment:
this probably isn't an issue but if `use_integrated_lists=True`, won't the extra `banned_subwords` be added twice? Once here and again inside `download_data`

Reply:
that's probably fine, if the trie already exists add_word should just return False